src/Service/MapService.php

```diff
@@ -0,0 +1,79 @@
+<?php declare(strict_types=1);
+
+
+namespace App\Service;
+
+
+use App\Dto\MapDTO;
+use App\Entity\Place;
+use App\Vo\CoordinatesVO;
+use App\Vo\PolygonVO;
+use Exception;
+
+class MapService
+{
+    /** Create MapDTO if at least one place has full coordinates, otherwise return null.
+     *
+     * @param Place[] $places
+     *
+     * @return ?MapDTO
+     */
+    public function createMapDTO(array $places): ?MapDTO
+    {
+        $map = null;
+        $points = [];
+        foreach ($places as $place) {
+            if ($place->getLat() !== null && $place->getLon() !== null){
+                $coordinates = new CoordinatesVO($place->getLat(), $place->getLon());
```

Comment:
Place->getCoordinates() return  ?CoordinatesVO

Reply:
Do you want add this method to Place entity?

Reply:
Yes